tests/devices/unit_tests/oav/test_snapshots.py

```diff
@@ -0,0 +1,177 @@
+    mock_beam_x = create_and_set_mock_signal_r(int, "moxk_beam_x", 510)
+    mock_beam_y = create_and_set_mock_signal_r(int, "mock_beam_y", 380)
+    async with DeviceCollector(mock=True):
+        snapshot = SnapshotWithBeamCentre("", mock_beam_x, mock_beam_y, "fake_snapshot")
+    set_mock_value(snapshot.directory, "/tmp/")
+    set_mock_value(snapshot.filename, "test")
+    set_mock_value(snapshot.url, "http://test.url")
+    return snapshot
+
+
+@pytest.fixture
+async def grid_snapshot() -> SnapshotWithGrid:
+    async with DeviceCollector(mock=True):
+        grid_snapshot = SnapshotWithGrid("", "fake_grid")
+
+    set_mock_value(grid_snapshot.top_left_x, 100)
+    set_mock_value(grid_snapshot.top_left_y, 100)
+    set_mock_value(grid_snapshot.box_width, 50)
+    set_mock_value(grid_snapshot.num_boxes_x, 15)
+    set_mock_value(grid_snapshot.num_boxes_y, 10)
+
+    set_mock_value(grid_snapshot.directory, "/tmp/")
+    set_mock_value(grid_snapshot.filename, "test")
+    set_mock_value(grid_snapshot.url, "http://test.url")
+    return grid_snapshot
+
+
+@patch("dodal.devices.areadetector.plugins.MJPG.Image")
+@patch("dodal.devices.oav.snapshots.snapshot_with_beam_centre.ImageDraw")
+@patch(
+    "dodal.devices.areadetector.plugins.MJPG.ClientSession.get",
+    autospec=True,
+)
+async def test_snapshot_with_beam_centre_triggered_then_crosshair_drawn_and(
+    mock_get, patch_image_draw, patch_image, snapshot
+):
+    mock_get.return_value.__aenter__.return_value = (mock_response := AsyncMock())
+    mock_response.ok = True
+    mock_response.read.return_value = (test_data := b"TEST")
+
+    patch_line = MagicMock()
+    patch_image_draw.Draw.return_value.line = patch_line
+
+    snapshot._save_image = (mock_save := AsyncMock())
+
+    mock_open = patch_image.open
+    mock_open.return_value.__aenter__.return_value = test_data
+
+    await snapshot.trigger()
+
+    assert len(patch_line.mock_calls) == 2
+    mock_save.assert_awaited_once()
+
+
+@patch("dodal.devices.areadetector.plugins.MJPG.Path.mkdir")
+@patch("dodal.devices.areadetector.plugins.MJPG.Image")
+@patch(
+    "dodal.devices.areadetector.plugins.MJPG.ClientSession.get",
+    autospec=True,
+)
+@patch("dodal.devices.areadetector.plugins.MJPG.aiofiles", autospec=True)
+async def test_snapshot_with_beam_centre_correctly_triggered_and_saved(
+    mock_aiofiles, mock_get, patch_image, mock_mkdir, snapshot
+):
+    mock_get.return_value.__aenter__.return_value = (mock_response := AsyncMock())
+    mock_response.ok = True
+    mock_response.read.return_value = (test_data := b"TEST")
+
+    mock_aio_open = mock_aiofiles.open
+    mock_aio_open.return_value.__aenter__.return_value = (mock_file := AsyncMock())
+
+    mock_open = patch_image.open
+    mock_open.return_value.__aenter__.return_value = test_data
+
+    # Set new directory and test that it's created
+    set_mock_value(snapshot.directory, "new_dir")
+
+    await snapshot.trigger()
+
+    mock_mkdir.assert_called_once()
```

Comment:
Nit: I would test this in a separate test that's just testing a new folder is created i.e. like `test_given_directory_not_existing_when_snapshot_triggered_then_directory_created`